internal/versions/versions.go

```diff
@@ -168,6 +168,9 @@ func (h *handler) GetDefaultReleaseImage(cpuArchitecture string) (*models.Releas
 
 // Returns the OsImage entity
 func (h *handler) GetOsImage(openshiftVersion, cpuArchitecture string) (*models.OsImage, error) {
+	if cpuArchitecture == common.MultiCPUArchitecture {
```

Comment:
So IIUC, we don't search for an OS image here as the work on multiarch release images is still work in progress, so we skip any validations. Can you maybe write some comment about it here to make clear? Just cause it seems a bit weird we just early return in this case.

Reply:
>so we skip any validations

That's not completely true. It's more that we are moving them to the later stage. As I described in the PR, when you use a regular single-arch release image, the validation of having OS image available happens immediately when you create your Cluster object (e.g. via AgentClusterInstall)

Now, if you use multiarch in ClusterImageSet, you can create AgentClusterInstall for whatever, because at this moment no one knows what architecture you are going to use (it will be defined only in InfraEnv). So Cluster object itself is not validating whether OS image exists, because we don't know which OS image to look for (we know version, but not arch).

You may ask, shouldn't we require all the architecture from the multiarch release image to exist? The answer is no, because release image contains lot more architectures than we have OS images for, thus we don't want to require a customer who wants to install e.g. only ARM using multiarch release image to also have OS image for powerpc.

You may ask, if Cluster does not validate that we have OS image, who does? The answer is InfraEnv. This is the only component in the whole chain that knows what architecture you want. So when InfraEnv tries to create an ISO, it will know the OCP version and will know the architecture. At this moment if we have a matching OS image we will proceed as usual. If during ISO generation we realize that OS image is missing for your version&arch, you will get a clear message `"Failed to create image: No OS image for Openshift version 4.11.0-0.nightly-multi-2022-07-26-151412 and architecture XYZ"`

I hope that makes it clearer. Not sure how much of this you'd like me to put directly into the code as comments, so I'm open for suggestions.

Reply:
Ok, added a comment with a short summary of the above